libsolidity/AST.h

```diff
@@ -1112,13 +1112,22 @@ class ElementaryTypeNameExpression: public PrimaryExpression
 };
 
 /**
- * A literal string or number. @see Type::literalToBigEndian is used to actually parse its value.
+ * A literal string or number. @see ExpressionCompiler::endVisit() is used to actually parse its value.
  */
 class Literal: public PrimaryExpression
 {
 public:
-	Literal(Location const& _location, Token::Value _token, ASTPointer<ASTString> const& _value):
-		PrimaryExpression(_location), m_token(_token), m_value(_value) {}
+	enum class SubDenomination
+	{
+		NONE = Token::ILLEGAL,
```

Comment:
Gavin this is part of already existing Solidity Tokens. I am not going to change it with this PR. We are going to schedule one PR to change them all to Camelcase. For this PR I am just trying to make sure not to introduce any new incosistencies.